File: lang/nodejs/test/fun/vac/data_structures/ArrayQueueTest.js
"use strict"

const ArrayQueue = require("fun/vac/data_structures/ArrayQueue")
const TestRunner = require("fun/vac/util/TestRunner")

const testArrayQueue = () => {
    let size = 8192

    const queue = new ArrayQueue.ArrayQueue(0)

    for (let i = 1; i <= size; i++) {
        queue.enqueue(i)
    }

    queue.shrink()

    if (queue.size() !== size) {
        return false
    }

    if (queue.capacity() !== size) {
        return false
    }

    for (let i = 1; i <= size; i++) {
        if (queue.peek() !== i) {
            return false
        }
        queue.dequeue()
    }

    queue.shrink()

    if (!queue.isEmpty()) {
        return false
    }

    if (queue.capacity() !== 0) {
        return false
    }

    return true
}

if (module === require.main) {
    TestRunner.parseTest(testArrayQueue())
}

File: lang/python/test/fun/vac/data_structures/ArrayQueueTest.py
from fun.vac.data_structures.ArrayQueue import ArrayQueue
from fun.vac.util import TestRunner

def testArrayQueue()-> bool:
    size = 8192

    queue = ArrayQueue(0)

    for i in range(1, size + 1):
        queue.enqueue(i)

    queue.shrink()

    if queue.size() != size:
        return False

    if queue.capacity() != size:
        return False

    for i in range(1, size + 1):
        if queue.peek() != i:
            return False
        queue.dequeue()

    queue.shrink()

    if not queue.isEmpty():
        return False

    return queue.capacity() == 0

if __name__ == "__main__":
    TestRunner.parseTest(testArrayQueue())
